Handle unoptimized branches in CodeFolding

tlively opened the pull request:
CodeFolding previously did not consider br_on_* instructions at all, so
it would happily merge tails even if there were br_on_* branches to the
same label with non-matching tails. Fix the bug by making any label
targeted by any instruction not explicitly handled by CodeFolding
unoptimizable. This will gracefully handle other branching instructions
like `resume` and `resume_throw` as well. Folding these branches
properly is left as future work.

Also rename the test file from code-folding_enable-threads.wast to just
code-folding.wast and enable all features instead of just threads. The
old name was left over from when the test was originally ported to lit,
and the new feature is necessary because the new test uses GC
instructions.

kripken commented on src/passes/CodeFolding.cpp:
```diff
@@ -155,6 +155,14 @@ struct CodeFolding : public WalkerPass<ControlFlowWalker<CodeFolding>> {
+
   void visitSwitch(Switch* curr) {
     for (auto target : curr->targets) {
       unoptimizables.insert(target);
```
Another option could be to use `visitExpression` and `BranchUtils` to rule out all branches but a normal `Break`, which would avoid special-casing `BrOn` and `Switch`. But lgtm as that might not be worth it.

tlively replied:
That's a good idea. That will handle `resume` and `resume_throw` as well.

tlively commented:
Updated the implementation and rebased this onto main so it can land before the more controversial change that prevents concrete If arms from being folded.
